src/cryptonote_core/tx_pool.cpp

```diff
@@ -235,34 +237,15 @@ bool tx_memory_pool::have_duplicated_non_standard_tx(
                 return true;
             }
         }
-    } else if (tx.type == txtype::ethereum_new_service_node) {
-        cryptonote::tx_extra_ethereum_new_service_node data = {};
-        if (!cryptonote::get_field_from_tx_extra(tx.extra, data)) {
+    } else if (is_l2_event_tx(tx.type)) {
+        std::string fail;
+        auto event = eth::extract_event(hard_fork_version, tx, fail);
```

Comment:
Are we missing a check to ensure that the L2 event doesn't already exist in the txpool? The other branches do this kind of business

```cpp
        std::vector<transaction> pool_txs;
        get_transactions(pool_txs);
        for (const transaction& pool_tx : pool_txs) {
            if (pool_tx.type != txtype::state_change)
                continue;
```

Reply:
Also this check might need to be more robust with "double spends", or for instance yesterday's situation by hoisting out 

```cpp
        if (m_db->tx_exists(tx_id)) {
            log::info(
                    logcat,
                    fg(fmt::terminal_color::red),
                    "Block with id: {} attempting to add transaction already in blockchain with "
                    "id: {}",
                    id,
                    tx_id);
            return false;
        }
```

Seems it's possible for a duplicate TX hash to be put into the TX pool. Once it's in it stalls the network until someone coordinates a flush of the pool? Or it expires (can't remember if that happens or not) But if we block dupe TX's at the TX pool level this should be impossible to do at all.

I can't tell if we check for dupe TX hashes for other transactions, I see we have dupe key-image checks but our custom TX types without key-images in general could benefit from also having this check hoisted up earlier in the function to eliminate that issue.

Reply:
Yes, I'll add it.  I thought that it would just fail if it already exists, but it looks like it removes it and then reinserts it which seems unnecessary.

Reply:
Added.

It doesn't look like we ever check for identical tx hashes, but I don't think we need to because the other checks being done are stronger: e.g. obligation quorum state change txes could have different subsets of the validator signatures for the same state change (and thus different hashes, even though it is the exact same change); and a regular tx could be spending the same inputs but include different random components.  In each case an *actually* identical hash is just a subset of the possible duplicate transactions.